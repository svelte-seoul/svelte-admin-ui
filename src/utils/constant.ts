export const MOBILE_SIZE = 640;