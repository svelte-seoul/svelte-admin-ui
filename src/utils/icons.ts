import svgHamburger from '../../assets/svgs/hamburger.svg';
import svgLock from '../../assets/svgs/lock.svg';
import svgLogo from '../../assets/svgs/logo.svg';
import svgMail from '../../assets/svgs/mail.svg';

export {svgLogo, svgMail, svgLock, svgHamburger};