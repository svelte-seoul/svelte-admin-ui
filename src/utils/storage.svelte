<script context="module" lang="ts">
  const getStorage = (key: string): string | null => {
    return localStorage.getItem(key);
  };

  const setStorage = (key: string, value: string): void => {
    return localStorage.setItem(key, value);
  };

  const destroyStorage = (key: string): void => {
    return localStorage.removeItem(key);
  };

  const getSessionStorage = (key: string): string | null => {
    return sessionStorage.getItem(key);
  };

  const setSessionStorage = (key: string, value: string): void => {
    return sessionStorage.setItem(key, value);
  };

  const destroySessionStorage = (key: string): void => {
    return sessionStorage.removeItem(key);
  };

  export {
    getStorage,
    setStorage,
    destroyStorage,
    getSessionStorage,
    setSessionStorage,
    destroySessionStorage,
  };
</script>
