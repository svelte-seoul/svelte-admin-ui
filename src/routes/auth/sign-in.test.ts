import {destroyStorage, getStorage, setStorage} from '../../utils/storage';

import SignIn from './sign-in.svelte';
import type {ThemeStore} from 'svelte-theme';
import { getContext } from 'svelte';
import {render} from '@testing-library/svelte';
import {setUpUser} from '../../../test/testUtils';

describe('render test', () => {
    it('should render without crashing', () => {
        const testingLib = render(SignIn);

        expect(testingLib).toMatchSnapshot();
    });
});

describe('interaction test', () => {
    beforeEach(() => {
        destroyStorage('token');
    });
    
    it('should login when email or password are typed', async () => {
        const {user, getByPlaceholderText, getAllByRole} = setUpUser(SignIn);
    
        const emailInput = getByPlaceholderText('email@email.com');
        const passwordInput = getByPlaceholderText('********');
        const loginButton = getAllByRole('button')[0];

        await user.type(emailInput, 'test');
        await user.type(passwordInput, 'psw');

        await user.click(loginButton);

        expect(getStorage('token')).toBe('temp token')
    });

    it('should redirect to main page when a user already logged in', () => {
        const mockFn = jest.fn();
        window.alert = mockFn; 
        setStorage('token', 'test token');

        render(SignIn);

        expect(mockFn).toBeCalled();
    });
    
    it('should change theme type when a user types [ctrlKey] + .', async () => {
        const {user, getByTestId} = setUpUser(SignIn);

        const container = getByTestId('sign-in-container');

        const currentBackgroundColor = window.getComputedStyle(container);

        await user.keyboard('{ctrlKey>}{.}');

        expect(currentBackgroundColor).not.toEqual(window.getComputedStyle(container))
    });
});
