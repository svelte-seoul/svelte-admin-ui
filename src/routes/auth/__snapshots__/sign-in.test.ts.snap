// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test should render without crashing 1`] = `
Object {
  "component": Sign_in {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        [Function],
        false,
        "",
        "",
        [Function],
        [Function],
        [Function],
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
        [Function],
        [Function],
        [Function],
        [Function],
      ],
      "on_disconnect": Array [],
      "on_mount": Array [],
      "props": Object {
        "handleLogin": 0,
      },
      "root": <div>
        <div
          class="container svelte-5hdqa"
          data-testid="sign-in-container"
        >
          <img
            alt="logo"
            class="logo svelte-5hdqa"
            src="[object Object]"
          />
           
          <h3
            class="svelte-5hdqa"
          >
            User Program
          </h3>
           
          <div
            class="edit-text svelte-xguohu"
            data-testid="container"
            style=""
          >
            <div
              class="container svelte-xguohu"
            >
              <img
                alt="email"
                class="icon svelte-5hdqa"
                slot="leftElement"
                src="[object Object]"
              />
               
              <input
                alt="alt-input"
                class="svelte-xguohu"
                placeholder="email@email.com"
                style=""
                type="email"
              />
               
              <div />
            </div>
             
          </div>
           
          <div
            class="edit-text svelte-xguohu"
            data-testid="container"
            style=""
          >
            <div
              class="container svelte-xguohu"
            >
              <img
                alt="lock"
                class="icon svelte-5hdqa"
                slot="leftElement"
                src="[object Object]"
              />
               
              <input
                alt="alt-input"
                class="svelte-xguohu"
                placeholder="********"
                style=""
                type="password"
              />
               
              <div />
            </div>
             
          </div>
           
          <button
            class="button svelte-dp9g4h primary medium"
            style=""
          >
            <div
              class="text-contrast svelte-5hdqa"
            >
              Login
            </div>
          </button>
           
          <button
            class="button svelte-dp9g4h medium"
            style=""
          >
            <div
              data-testid="text"
            >
              Sign Up
            </div>
            
          </button>
           
          <button
            class="button svelte-dp9g4h transparent medium"
            style="margin-top: -2px; font-size: 14px; text-decoration: underline;"
          >
            <div
              data-testid="text"
            >
              Look around
            </div>
            
          </button>
        </div>
      </div>,
      "skip_bound": false,
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div>
      <div
        class="container svelte-5hdqa"
        data-testid="sign-in-container"
      >
        <img
          alt="logo"
          class="logo svelte-5hdqa"
          src="[object Object]"
        />
         
        <h3
          class="svelte-5hdqa"
        >
          User Program
        </h3>
         
        <div
          class="edit-text svelte-xguohu"
          data-testid="container"
          style=""
        >
          <div
            class="container svelte-xguohu"
          >
            <img
              alt="email"
              class="icon svelte-5hdqa"
              slot="leftElement"
              src="[object Object]"
            />
             
            <input
              alt="alt-input"
              class="svelte-xguohu"
              placeholder="email@email.com"
              style=""
              type="email"
            />
             
            <div />
          </div>
           
        </div>
         
        <div
          class="edit-text svelte-xguohu"
          data-testid="container"
          style=""
        >
          <div
            class="container svelte-xguohu"
          >
            <img
              alt="lock"
              class="icon svelte-5hdqa"
              slot="leftElement"
              src="[object Object]"
            />
             
            <input
              alt="alt-input"
              class="svelte-xguohu"
              placeholder="********"
              style=""
              type="password"
            />
             
            <div />
          </div>
           
        </div>
         
        <button
          class="button svelte-dp9g4h primary medium"
          style=""
        >
          <div
            class="text-contrast svelte-5hdqa"
          >
            Login
          </div>
        </button>
         
        <button
          class="button svelte-dp9g4h medium"
          style=""
        >
          <div
            data-testid="text"
          >
            Sign Up
          </div>
          
        </button>
         
        <button
          class="button svelte-dp9g4h transparent medium"
          style="margin-top: -2px; font-size: 14px; text-decoration: underline;"
        >
          <div
            data-testid="text"
          >
            Look around
          </div>
          
        </button>
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
